tests/functional/generate_pages.py

```diff
@@ -227,9 +227,9 @@ def process_options(answer_id, options, page_spec, base_prefix):
 
         # Add a selector for an option which exposes another input, e.g.
         # an 'Other' option which exposes a text input for the user to fill in
-        if 'child_answer_id' in option:
+        if 'detail_answer' in option:
             option_context = {
-                'answerId': option['child_answer_id'],
+                'answerId': option['detail_answer']['id'],
                 'answerName': option_name + 'Text'
```

Comment:
Text might not make sense if it's a number or percentage. Maybe something generic like `Detail` would work.

Reply:
You're quite right you pain in the *!$%@. I'll update all the specs accordingly.